chemprop/features/featurization.py

```diff
@@ -178,57 +242,133 @@ def __init__(self, mol: Union[str, Chem.Mol],
         self.overwrite_default_atom_features = overwrite_default_atom_features
         self.overwrite_default_bond_features = overwrite_default_bond_features
 
-        # Get atom features
-        self.f_atoms = [atom_features(atom) for atom in mol.GetAtoms()]
-        if atom_features_extra is not None:
-            if overwrite_default_atom_features:
-                self.f_atoms = [descs.tolist() for descs in atom_features_extra]
-            else:
-                self.f_atoms = [f_atoms + descs.tolist() for f_atoms, descs in zip(self.f_atoms, atom_features_extra)]
-
-        self.n_atoms = len(self.f_atoms)
-        if atom_features_extra is not None and len(atom_features_extra) != self.n_atoms:
-            raise ValueError(f'The number of atoms in {Chem.MolToSmiles(mol)} is different from the length of '
-                             f'the extra atom features')
-
-        # Initialize atom to bond mapping for each atom
-        for _ in range(self.n_atoms):
-            self.a2b.append([])
-
-        # Get bond features
-        for a1 in range(self.n_atoms):
-            for a2 in range(a1 + 1, self.n_atoms):
-                bond = mol.GetBondBetweenAtoms(a1, a2)
-
-                if bond is None:
-                    continue
-
-                f_bond = bond_features(bond)
-                if bond_features_extra is not None:
-                    descr = bond_features_extra[bond.GetIdx()].tolist()
-                    if overwrite_default_bond_features:
-                        f_bond = descr
+        if not REACTION_MODE:
```

Comment:
REACTION_MODE is not a Boolean is it?

Reply:
Same as above, the check works out, since `bool(None)=False` and `bool("anystring")=True` which is correct but ugly. Will correct this to `if REACTION_MODE == None`

Reply:
Due to the new treatment of the global variables, this is now nicer, too!